gst/droidcamsrc/gstdroidcamsrc.c

```diff
@@ -354,24 +355,25 @@ gst_droidcamsrc_set_property (GObject * object, guint prop_id,
       g_mutex_unlock (&src->capture_lock);
 
       /* deactivate old mode */
-      if (src->active_mode) {
+      was_active = (src->active_mode != NULL);
+      if (was_active) {
```

Comment:
I think introducing this boolean is a bit unnecessary. It would be cleaner to keep the old if, include the whole block in it, then duplicate the src->mode setting into an else.

Reply:
I've changed the code so that the extra boolean isn't needed. Please take another look.

Reply:
Ah good, that's even cleaner!